build-all.sh

```diff
@@ -87,6 +86,16 @@ while read -r PKG PKG_DIR; do
 
 	# Update build status
 	echo "$PKG" >> "$BUILDSTATUS_FILE"
+
+	# Check which packages were also compiled
+	if [ -z "$TERMUX_INSTALL_DEPS" ]; then
+		for build_pkg in ~/.termux-build/*; do
+                        pkgname="${build_pkg##*/}"
+			(grep -q '^_' <<< "${pkgname}" || grep -q "^${pkgname}\$" "$BUILDSTATUS_FILE") && continue
+			echo "The \"${pkgname}\" package was also compiled"
+			echo "${pkgname}" >> "$BUILDSTATUS_FILE"
```

Comment:
I don't think this is a good idea. If build environment is not clean then the previously built packages will be added to buildstatus by this. That is most likely not what the builder wants. 

I.e. if I run:

```
./build-all.sh -o ./special-build-dir/
# .... ctrl-c after X packages
./build-package.sh some-other-package
./build-all.sh -o ./special-build-dir/
```
Then some-other-package foo will be added to buildstatus file, but it will not be in special-build-dir/. I think it is reasonable for build-all.sh to build all packages, and not add extra packages it finds to the buildstatus file.

Reply:
I think keeping the build status file would be very good to keep around as a commandline option.

I've been using that extensively as part of the #21130 build testing.
- Run `./build-all.sh`
- Some package errors
- Read the log, add it manually to the end of `buildstatus.txt` to skip it
- Repeat

Reply:
> I don't think this is a good idea. If build environment is not clean then the previously built packages will be added to buildstatus by this. That is most likely not what the builder wants.
> 
> I.e. if I run:
> 
> ```
> ./build-all.sh -o ./special-build-dir/
> # .... ctrl-c after X packages
> ./build-package.sh some-other-package
> ./build-all.sh -o ./special-build-dir/
> ```
> 
> Then some-other-package foo will be added to buildstatus file, but it will not be in special-build-dir/.

This example I absolutely agree with.
I've had to do a couple spot checks for individual packages out of order.
Running something like `./build-package.sh coreutils` should not add to the `buildstatus.txt` file.

Reply:
This check allows you to check if other packages have been compiled while other packages are being compiled. This mostly happens with package dependencies that have a circular dependency with the package. Thanks to this check, `build-all.sh` avoids recompiling the package.

> I don't think this is a good idea. If build environment is not clean then the previously built packages will be added to buildstatus by this. That is most likely not what the builder wants.

Thanks for noting. ~~I think that in the script `build-all.sh` I need to add the process of cleaning the environment (i.e. running the script `clean.sh`) before compiling all the packages.~~ I'll think about how to solve this.